### PR TITLE
iphone2g wip (nw)

### PR DESCRIPTION
**Pull request by MoochMcGee**

### CODE REVIEW

File: src/devices/cpu/arm7/arm7.cpp [hunk 1]
```diff
@@ -1778,6 +1782,127 @@ void arm7_cpu_device::arm7_dt_w_callback(uint32_t insn, uint32_t *prn)
...
+	uint8_t cpnum = (opcode & INSN_COPRO_CPNUM) >> INSN_COPRO_CPNUM_SHIFT;
+	uint32_t data = 0;
+
+//  printf("arm7946: copro %d write %x to cReg %d op2 %d op3 %d (mask %08x)\n", cpnum, data, cReg, op2, op3, mem_mask);
+
+	if (cpnum == 15)
+	{
+		switch (crn)
+		{
+			case 0:
```
Comment (angelosa) [inline src/devices/cpu/arm7/arm7.cpp]: Way too much indentation here and its sister write callback below, just use IFs for now.

File: src/devices/machine/vic_pl192.cpp [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        else m_out_irq_func(0);
+    }
+}
+
+uint32_t vic_pl192_device::read(offs_t offset)
+{
+    offset &= 0xffc;
+    switch(offset)
+    {
+        case 0x000:
```
Comment (angelosa) [inline src/devices/machine/vic_pl192.cpp]: Device should be a device_memory_interface with these offsets into an address_map. Will save some typing and indentation.

Comment (MoochMcGee) [reply src/devices/machine/vic_pl192.cpp]: I honestly don't see how that would save on typing, but okay...

File: src/devices/cpu/arm7/arm7.cpp (hunk 1 above)
Comment (MoochMcGee) [reply src/devices/cpu/arm7/arm7.cpp]: Done

File: src/devices/machine/vic_pl192.cpp (hunk 2 above)
Comment (MoochMcGee) [reply src/devices/machine/vic_pl192.cpp]: Oh well, I did it your way as requested.

**Comment by MoochMcGee** [comment]
All previous feedback has been addressed.

**Comment by rb6502** [comment]
Note that the iPhone 2G/3G SoC is believed to be either a Samsung S3C2460 or S3C6400.  If it's a 2460, the on-board peripheral emulation in devices/machine/s3c24xx.cpp should be used rather than duplicating them here.  I'm going to apply this for now anyway though.

**Comment by MoochMcGee** [comment]
@rb6502 It's not though, it's an S5L8900.

**Comment by Tafoid** [comment]
This commit doesn't pass the sniff test as it does not pass -validation check.  Could you look at correcting this, please?

> c:\mamegit\mame>mame64 -validate
3 errors, 0 warnings
Errors:
   Device iphone2g.cpp(iphone2g_spi) short name is a duplicate of iphone2g.cpp(iphone2g_spi)
   Device iphone2g.cpp(iphone2g_spi) name 'iPhone 2G SPI controller' is a duplicate of iphone2g.cpp(iphone2g_spi)
   Device iphone2g.cpp(iphone2g_spi) reports type '19iphone2g_spi_device' (created with '21iphone2g_timer_device')

> Validity check failed (3 errors, 0 warnings in total)

**Pull request merged**